src/template.ts

```diff
@@ -1,6 +1,11 @@
 // we only support the simplest of template expressions
 const templateExpression = /({[a-zA-Z0-9_[\].]+})/g;
 const templateExpressionVarName = /{([a-zA-Z0-9_[\].]+)}/g;
+// We have a special keyword that we use to expand out an array for a chat_history argument
+const CHAT_HISTORY = "chat_history";
+const CHAT_HISTORY_VARIABLE = "{chat_history}";
```

Comment:
the challenge here is that you can have multiple variables, i.e. `chat_history` is not a unique name. You might have `{support_history}` `{rag_history}` or whatever